### PR TITLE
feat: support glob expansion

### LINKED ISSUE #6: Support glob expansion
- globstar
- ~~nullglob? Maybe yes?~~ error on no matches

### PR DESCRIPTION
**Pull request by dsherret**
* Uses the https://crates.io/crates/glob crate
* Errors when a glob doesn't match anything 

Closes #6

### CODE REVIEW

File: src/shell/test.rs [hunk 1]
```diff
@@ -976,6 +976,194 @@ pub async fn custom_command() {
...
 
+#[tokio::test]
+async fn glob_basic() {
+  TestBuilder::new()
+    .file("test.txt", "test\n")
+    .file("test2.txt", "test2\n")
+    .command("cat *.txt")
+    .assert_stdout("test\ntest2\n")
+    .run()
+    .await;
```
Comment (bartlomieju) [inline src/shell/test.rs:987]: Great tests 👍 this pattern works so well

File: src/shell/mod.rs [hunk 2]
```diff
@@ -322,6 +326,12 @@ async fn resolve_redirect_pipe(
     stderr.clone(),
   )
   .await;
+  let words = match words {
+    Ok(word) => word,
+    Err(err) => {
+      return Err(err.into_exit_code(stderr));
+    }
+  };
```
Comment (bartlomieju) [inline src/shell/mod.rs:334]: Nitpick: seems like `let/else` could work nicely in several places in this file.

File: src/shell/mod.rs [hunk 3]
```diff
@@ -570,93 +590,261 @@ async fn evaluate_args(
...
+            current_text.push_str(&text);
+          }
         }
+      }
+      let is_absolute = std::path::PathBuf::from(&current_text).is_absolute();
+      let cwd = state.cwd();
+      let pattern = if is_absolute {
+        current_text
+      } else {
+        format!("{}/{}", cwd.display(), current_text)
```
Comment (bartlomieju) [inline src/shell/mod.rs:709]: Does this work on windows?

File: src/shell/mod.rs (hunk 2 above)
Comment (dsherret) [reply src/shell/mod.rs:334]: do you know how to use let/else and get the error?

File: src/shell/mod.rs (hunk 3 above)
Comment (dsherret) [reply src/shell/mod.rs:709]: Yup. This is fine for the pattern.

File: src/shell/mod.rs (hunk 2 above)
Comment (bartlomieju) [reply src/shell/mod.rs:334]: Ah, good point. Disregard.

**Review by bartlomieju — APPROVED** [review]
LGTM, exciting!

**Pull request merged**